components/Buttons/tests/unit/FloatingButtonTests.m

```diff
@@ -488,13 +498,38 @@ - (void)testExpandedLayout {
                              CGRectGetMaxX(imageFrame) + button.imageTitleSpacing, 1);
 }
 
+- (void)testExpandedLayoutWithImageLocationTrailing {
+  // Given
+  MDCFloatingButton *button =
+      [[MDCFloatingButton alloc] initWithFrame:CGRectZero shape:MDCFloatingButtonShapeDefault];
+  button.mode = MDCFloatingButtonModeExpanded;
+  button.imageLocation = MDCFloatingButtonImageLocationTrailing;
+  [button setTitle:@"A very long title that can never fit" forState:UIControlStateNormal];
+  [button setImage:fakeImage() forState:UIControlStateNormal];
+  [button setMaximumSize:CGSizeMake(100, 48)
+                forShape:MDCFloatingButtonShapeDefault
+                  inMode:MDCFloatingButtonModeExpanded];
+  button.imageTitleSpacing = 12;
+
+  // When
+  [button sizeToFit];
+
+  // Then
+  CGRect imageFrame = button.imageView.frame;
+  CGRect titleFrame = button.titleLabel.frame;
+  XCTAssertEqualWithAccuracy(CGRectGetMaxX(imageFrame), CGRectGetWidth(button.bounds)-16, 1);
```

Comment:
Where does the -16 come from?

Reply:
It's from `internalLayoutSpacing.left` but in this case flipped for the `.trailing` imageLocation value.  It's not great that it's both here and in the implementation, but I didn't feel like it was worth externing the UIEdgeInsets in the implementation.